ludwig/schema/encoders.py

```diff
@@ -0,0 +1,134 @@
+from abc import ABC
+from typing import ClassVar, List, Optional, Union
+
+from marshmallow import Schema
+from marshmallow_dataclass import dataclass
+
+from ludwig.encoders.base import Encoder
+from ludwig.encoders.generic_encoders import DenseEncoder, PassthroughEncoder
+from ludwig.schema import utils as schema_utils
+
+
+@dataclass
+class BaseEncoderConfig(schema_utils.BaseMarshmallowConfig, ABC):
+    """Base class for encoders. Not meant to be used directly.
+
+    The dataclass format prevents arbitrary properties from being set. Consequently, in child classes, all properties
+    from the corresponding Encoder class are copied over.
+    """
+
+    encoder_class: ClassVar[Optional[Encoder]] = None
+    "Class variable pointing to the corresponding Encoder class."
+
+    encoder: str
+    "Name corresponding to an encoder."
+
+
+@dataclass
+class DenseEncoder(Schema):
+    """DenseEncoder is a dataclass that configures the parameters used for a dense encoder."""
+
+    encoder_class: ClassVar[Encoder] = DenseEncoder
+
+    type: str = "dense"
+
+    fc_layers: Optional[List[dict]] = schema_utils.DictList(
+        default=None,
+        description="List of fully connected layers to use in the encoder.",
+    )
+
+    num_layers: Optional[int] = schema_utils.PositiveInteger(
+        default=1,
+        description="Number of stacked fully connected layers that the input to the feature passes through.",
+    )
+
+    output_size: Optional[int] = schema_utils.PositiveInteger(
+        default=256,
+        description="Size of the output of the feature.",
+    )
+
+    use_bias: Optional[bool] = schema_utils.Boolean(
+        default=True,
+        description="Whether the layer uses a bias vector.",
+    )
+
+    weights_initializer: Optional[Union[str, dict]] = schema_utils.StringOptions(  # TODO: Add support for String/Dict
+        [
+            "constant",
```

Comment:
Could these options be replaced with `initializer_registry.keys()`

Reply:
Yes good call. These files will actually be used in a followup PR for the encoder schema refactor since the current implementation is not actually using this file. I will include the registry in the next PR however.

Reply:
Going to remove this file from this PR since it is not relevant until the encoder schema PR.